components/arena/python/arena/_arena_dist_tf.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+import datetime
+import logging
+
+# def DistributeTFOp(name, image, gpus: int, ):
+
+class DistributeTFOp(dsl.ContainerOp):
```

Comment:
```
def arena_distribute_tf_op(....):
    return MPIOp(....
    )
```

Reply:
Why is RandomNumOp? May I know what's the principle?   https://github.com/kubeflow/pipelines/blob/master/samples/basic/condition.py#L20

```
class RandomNumOp(dsl.ContainerOp):
  """Generate a random number between low and high."""
```

Reply:
Sorry for the confusing. Those samples are the oldest and have not been updated for a while.
Other samples are a bit more modern: https://github.com/kubeflow/pipelines/blob/master/samples/kubeflow-tf/kubeflow-training-classification.py

Your code (ineriting from `ContainerOp` vs returning `ContainerOp`) is not wrong. It's just outdated a bit.

Reply:
Thank you, I will update with your suggestions.